File: scratch/find_data.py
import os

search_dir = "c:/Users/ASUS/Desktop/LeeGo/TẠO APP - UP 1/leegoup1"
terms = ["InteractiveGame", "practiceQuestions", "curriculum", "Lesson 3", "checkUp"]

for root, dirs, files in os.walk(search_dir):
    if "node_modules" in root or ".git" in root or "scratch" in root:
        continue
    for file in files:
        if file.endswith((".ts", ".tsx", ".json")):
            path = os.path.join(root, file)
            try:
                with open(path, "r", encoding="utf-8") as f:
                    content = f.read()
                    found = [t for t in terms if t in content]
                    if found:
                        print(f"{file}: found {found}")
            except Exception as e:
                pass

File: scratch/find_data.js
const fs = require('fs');
const path = require('path');

const searchDir = "c:/Users/ASUS/Desktop/LeeGo/TẠO APP - UP 1/leegoup1";
const terms = ["InteractiveGame", "practiceQuestions", "curriculum", "Lesson 3", "checkUp"];

function walk(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        const stat = fs.statSync(fullPath);
        if (stat.isDirectory()) {
            if (file === 'node_modules' || file === '.git' || file === 'scratch') continue;
            walk(fullPath);
        } else if (file.endsWith('.ts') || file.endsWith('.tsx') || file.endsWith('.json')) {
            try {
                const content = fs.readFileSync(fullPath, 'utf8');
                const found = terms.filter(t => content.includes(t));
                if (found.length > 0) {
                    console.log(`${file}: found [${found.join(', ')}]`);
                }
            } catch (e) {}
        }
    }
}

walk(searchDir);
